Doc/library/typing.rst

```diff
@@ -1932,6 +1963,28 @@ Functions and decorators
    runtime we intentionally don't check anything (we want this
    to be as fast as possible).
 
+.. function:: assert_never(arg, /)
+
+   Statically assert that a line of code is unreachable.
```

Comment:
Oh, I'm guessing that "statically assert X" means "ask the type checker to verify that X is provable". If this is standard terminology, maybe this phrase should be explained somewhere?

Reply:
Well, C++ has a `static_assert()` function. But it's better to avoid the term if it's confusing, so I'll reword to "assert to the type checker".